GitCommands/StringExtensions.cs

```diff
@@ -189,6 +190,18 @@ public static string Quote(this string? s, string q = "\"")
             return string.IsNullOrEmpty(s) ? s : s.Quote();
         }
 
+        /// <summary>
+        /// Quotes and escapes this string for use as a command line argument.
+        /// </summary>
+        [Pure]
+        public static string QuoteForCommandLine(this string s, bool? forWindows = null)
+        {
+            return $"\"{((forWindows ?? EnvUtils.RunningOnWindows()) ? EscapeForWindowsCommandLine(s) : EscapeForPosixCommandLine(s))}\"";
```

Comment:
Curious: On what platform do you run GE if not Windows?

The quoting could theoretically be used with a WSL executable though.

Reply:
GE can run on Posix, as enforced by the existence and use of `EnvUtils.RunningOnWindows`. Notably, the check is currently used to choose between the console emulator (Windows) and the pipe based edit box terminal (Posix).